src/runtime/step_runner.js

```diff
@@ -8,7 +8,7 @@ const {beginTiming, endTiming} = Time
 
 async function run({attachmentManager, defaultTimeout, scenarioResult, step, stepDefinition, transformLookup, world}) {
   beginTiming()
-  const parameters = stepDefinition.getInvocationParameters({scenarioResult, step, transformLookup})
+  const parameters = await Promise.all(stepDefinition.getInvocationParameters({scenarioResult, step, transformLookup}))
```

Comment:
`getInvocationParameters` returns an array where each element is returned from one of the registered `Transform#transform` functions, which is userland code. If a user wants to write a transform that returns a `Promise`, Cucumber should be able to deal with that. I've given an example in cucumber/cucumber#108.

`Promise.all` takes an array of `Promise` objetcs and returns a new `Promise` that resolves to a new array of resolved values. If one of the passed in `Promise` objetcs is not a `Promise`, then it's wrapped in a `Promise#resolve`, so both async and regular transform functions will work.